src/AlgebraicGeometry/Schemes/ProjectiveVariety/Objects/Constructors.jl

```diff
@@ -0,0 +1,82 @@
+########################################################
+# (1) Generic constructors
+########################################################
+@doc raw"""
+    projective_variety(X::AbsProjectiveScheme; check::Bool=true) -> ProjectiveVariety
+
+Convert ``X`` to an projective variety.
+
+If check is set, then compute the reduced scheme of `X` first.
+"""
+function projective_variety(X::AbsProjectiveScheme{<:Field}; check::Bool=true)
+  check  ||  return ProjectiveVariety(X, check=check)
+  Xred,_ = reduced_scheme(X)
+  return ProjectiveVariety(Xred, check=check)
+end
+
+@doc raw"""
+    projective_variety(I::MPolyIdeal; check::Bool=true) -> ProjectiveVariety
+
+Return the projective variety defined by the homogeneous prime ideal ``I``.
+
+Since our varieties are irreducible, we check that ``I`` stays prime when
+viewed over the algebraic closure. This is an expensive check that can be disabled.
+"""
+projective_variety(I::MPolyIdeal{<:MPolyDecRingElem}; check::Bool=true) = ProjectiveVariety(ProjectiveScheme(base_ring(I),I), check=check)
+
+function projective_variety(R::MPolyDecRing, I::MPolyIdeal{<:MPolyDecRingElem}; check::Bool=true)
+  @req base_ring(I) === R "ideal must be defined over R"
+  ProjectiveVariety(ProjectiveScheme(R,I), check=check)
+end
+
+
+@doc raw"""
+    projective_variety(R::Ring; check::Bool=true)
+
+Return the projective variety defined by the ``\mathbb{Z}`` graded ring ``R``.
+
+We require that ``R`` is a finitely generated algebra over a field ``k`` and
+moreover that the base change of ``R`` to the algebraic closure ``\bar k``
+is an integral domain.
+"""
+projective_variety(R::Ring; check::Bool=true) = ProjectiveVariety(ProjectiveScheme(R), check=check)
+
+projective_variety(R::MPolyDecRing; check::Bool=true) = ProjectiveVariety(ProjectiveScheme(R), check=check)
+
+function projective_variety(f::MPolyDecRingElem; check=true)
```

Comment:
```suggestion
function projective_variety(f::MPolyDecRingElem{<:FieldElem}; check=true)
```
I suspect that the methods used below will not work unless we are working over an admissible field of coefficients. Maybe, we can relax to `Union{<:FieldElem, <:ZZRingElem}`.

Reply:
Since varieties are over a field anyways, adding `ZZRingElemen` won't give us much.